citar-latex.el

```diff
@@ -65,17 +65,58 @@ the point."
   (ignore-errors (reftex-get-bibfile-list)))
 
 ;;;###autoload
-(defun citar-latex-keys-at-point ()
-  "Return a list of keys at point in a latex buffer."
-  (unless (fboundp 'TeX-current-macro)
-    (error "Please install AUCTeX"))
-  (when (citar-latex-is-a-cite-command (TeX-current-macro))
-    (split-string (thing-at-point 'list t) "," t "[{} ]+")))
+(defun citar-latex-key-at-point ()
+  "Return citation key at point with its bounds.
+The return value is (KEY . BOUNDS), where KEY is the citation key
+at point and BOUNDS is a pair of buffer positions.  Returns nil
+if there is no key at point."
+  (save-excursion
+    (when-let* ((bounds (citar-latex--macro-bounds))
+                (keych "^,{}")
+                (beg (progn (skip-chars-backward keych (car bounds)) (point)))
+                (end (progn (skip-chars-forward keych (cdr bounds)) (point)))
+                (pre (buffer-substring-no-properties (car bounds) beg))
+                (post (buffer-substring-no-properties end (cdr bounds))))
+      (and (string-match-p "{\\([^{}]*,\\)?\\'" pre)  ; preceded by { ... ,
+           (string-match-p "\\`\\(,[^{}]*\\)?}" post) ; followed by , ... }
+           (goto-char beg)
+           (looking-at (concat "[[:space:]]*\\([" keych "]+?\\)[[:space:]]*[,}]"))
+           (cons (match-string-no-properties 1)
+                 (cons (match-beginning 1) (match-end 1)))))))
 
 ;;;###autoload
-(defun citar-latex-insert-keys (keys)
-  "Insert comma sperated KEYS in a latex buffer."
-  (insert (string-join keys ", ")))
+(defun citar-latex-citation-at-point ()
+  "Find citation macro at point and extract keys.
+Finds brace-delimited strings inside the bounds of the macro,
+splits them at comma characters, and trims whitespace.
+Returns (KEYS . BOUNDS), where KEYS is a list of the found
+citation keys and BOUNDS is a pair of buffer positions indicating
+the start and end of the citation macro."
```

Comment:
```suggestion
(defun citar-latex-citation-at-point ()
  "Find citation macro at point and extract keys.
  
Find brace-delimited strings inside the bounds of the macro,
splits them at comma characters, and trims whitespace.

Return (KEYS . BOUNDS), where KEYS is a list of the found
citation keys and BOUNDS is a pair of buffer positions indicating
the start and end of the citation macro."
```

Changed here to use the imperative consistently "return rather than returns".